raft: fix raft mixedversion forget_leader tests

iskettaneh opened the pull request:
Right now, the tests campaign a leader with a newer version (24.3). This means that followers with older versions (24.2) which have store liveness disabled would panic when receiving a MsgFortify from the leader.

Note that this scenario should not happen in reality because the leader would only send MsgFortify if it knows that everyone is at least at 24.3.

This commit fixes this by making all nodes have the same older version 24.2.

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/132699)
<!-- Reviewable:end -->

iskettaneh committed:
raft: fix raft mixedversion forget_leader tests

Right now, the tests campaign a leader with a newer version (24.3). This
means that followers with older versions (24.2) which have store
liveness disabled would panic when receiving a MsgFortify from the
leader.

Note that this scenario should not happen in reality because the leader
would only send MsgFortify if it knows that everyone is at least at
24.3.

This commit fixes this by making all nodes have the same older version
24.2.

Epic: None

Release note: None

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/132699#-:-O9Gns0Y4XYqP3Eqja8k:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @iskettaneh)



<!-- Sent from Reviewable.io -->

iskettaneh commented:
bors r+
